appinventor/components/src/com/google/appinventor/components/scripts/ComponentProcessor.java

```diff
@@ -629,6 +721,239 @@ protected String getRwString() {
     }
   }
 
+  /**
+   * An enum specifying the available types of helper blocks aka types of helper UI.
+   * Currently the only type of helper UI is an OPTION_LIST which defines a dropdown UI in the blocks
+   * editor. This is associated with the OptionList data type, ie OptionList data always has an
+   * OPTOIN_LIST style UI in the blocks editor (as of now).
+   */
+  protected enum HelperType { OPTION_LIST }
+
+  /**
+   * A key that allows you to access info about a helper block.
+   * 
+   * Currently HelperKeys are only used to define OptionList style helper blocks. Aka dropdown
+   * blocks for the current UI implementation in the blocks editor. In the future helper keys could
+   * be used to define other kinds of helper blocks like range blocks, default color blocks, etc.
+   */
+  protected final class HelperKey {
+    private HelperType helperType;
+
+    private String key;
+
+    /**
+     * Creates a HelperKey which can be used to access data about a helper block.
+     */
+    protected HelperKey(HelperType type, String key) {
+      this.helperType = type;
+      this.key = key;
+    }
+
+    /**
+     * Returns the type of helper block, aka the type of helper UI. Eg an option list.
+     * @return the type of helper block.
+     */
+    protected HelperType getType() {
+      return helperType;
+    }
+
+    /**
+     * Returns the key to the specific helper data. Eg in the case of an option list helper, this
+     * key could be used to look up values in the optionLists Map.
+     * @return key to the helper data.
+     */
+    protected String getKey() {
+      return key;
+    }
+  }
+
+
+  /**
+   * Represents a list of Options associated with some (enum) class. The data in this OptionList
+   * is used to create OptionList helper blocks.
+   * 
+   * Here JSON-ified example of such data, in this case we are looking at the Direction enum with
+   * a default value of East.
+   * {
+   *   "className": "com.google.appinventor.components.common.Direction",
+   *   "key": "Direction",
+   *   "tag": "Direction",
+   *   "defaultOpt": "East",
```

Comment:
This should now also include this extra line: 
      "underlyingType": "java.lang.Integer",

Reply:
[And done :D](https://github.com/BeksOmega/appinventor-sources/pull/3/commits/730fe3f1a503c9902701a360d2e2783769c5dc12)